packages/core-connector-interface/src/channel.d.ts

```diff
@@ -1,216 +1,50 @@
-      signature?: Signature
-    }
-  ): Promise<SignedChannel>
-
-  tickets: {
-    /**
-     * Submits a signed ticket to the blockchain.
-     * @param signedTicket a signed ticket
-     * @param secretA a signed ticket
-     * @param secretB a signed ticket
-     */
-    submit(
-      ticket: AcknowledgedTicket,
-      ticketIndex: Uint8Array
-    ): Promise<
-      | {
-          status: 'SUCCESS'
-          receipt: string
-          ackTicket: AcknowledgedTicket
-        }
-      | {
-          status: 'FAILURE'
-          message: string
-        }
-      | {
-          status: 'ERROR'
-          error: Error | string
-        }
-    >
-  }
+  new (indexer: Indexer, connector: any, self: Public, counterparty: Public): Channel
 }
 
 declare interface Channel {
-  // Id of the channel
-  readonly channelId: Promise<Hash>
-
-  // Timestamp once the channel can be settled
-  readonly settlementWindow: Promise<UINT256>
+  readonly counterparty: Public
 
-  // Current state counter of the channel
-  readonly stateCounter: Promise<UINT256>
+  getId(): Promise<Hash>
 
-  // Current status of the channel
-  readonly status: Promise<'CLOSED' | 'PENDING_TO_CLOSE' | 'OPEN'>
+  getState(): Promise<ChannelEntry>
 
-  // Current state of the channel, i.e. `FUNDED` with `1 HOPR / 3 HOPR`
-  readonly state: Promise<ChannelType>
+  getBalances(): Promise<{
+    self: Balance
+    counterparty: Balance
+  }>
 
-  // Current balance of partyA
-  readonly balance_a: Promise<Balance>
-  // Current balance of partyB
-  readonly balance_b: Promise<Balance>
+  open(fundAmount: Balance): Promise<void>
 
-  // Current total balance (sum of balance_a and balance_b)
-  readonly balance: Promise<Balance>
+  initializeClosure(): Promise<void>
+  finalizeClosure(): Promise<void>
 
-  readonly currentBalance: Promise<Balance>
+  createTicket(amount: Balance, challenge: Hash, winProb: number): Promise<SignedTicket>
 
-  readonly currentBalanceOfCounterparty: Promise<Balance>
+  createDummyTicket(challenge: Hash): Promise<SignedTicket>
 
-  readonly ticket: {
-    /**
-     * Constructs a ticket to use in a probabilistic payment channel.
-     * @param amount amount of funds to include
-     * @param challenge a challenge that has to be solved be the redeemer
-     * @param winProb probability for the generated ticket to be a win
-     */
-    create(
-      amount: Balance,
-      challenge: Hash,
-      winProb: number | undefined,
-      arr?: {
-        bytes: ArrayBuffer
-        offset: number
+  submitTicket(
+    ticket: AcknowledgedTicket,
+    ticketIndex: Uint8Array
+  ): Promise<
+    | {
+        status: 'SUCCESS'
+        receipt: string
+        ackTicket: AcknowledgedTicket
       }
-    ): Promise<SignedTicket>
-
-    /**
-     * Checks a previously issued ticket for its validity.
-     * @param signedTicket a previously issued ticket to check
-     * @param props additional arguments
-     */
-    verify(signedTicket: SignedTicket): Promise<boolean>
```

Comment:
this function was only used in tests, we do ticket checks in two areas:

- in core when we receive an acknowledgement ticket
- in core when we receive an unacknowledged ticket
- in core-ethereum before we submit a ticket